### PR TITLE
Kokkos resilience

### PR DESCRIPTION
**Pull request by jeffmiles63**
This is a DRAFT request...not intended to be merged any time soon...just need the PR for discussion ...

### CODE REVIEW

**Comment by crtrott** [comment]
One basic comment: this probably needs to eventually go into a different repository. I.e. the resilience stuff shouldn't live in Kokkos Core but something else.

File: core/src/Kokkos_View.hpp [hunk 1]
```diff
@@ -1985,7 +1985,7 @@ class View : public ViewTraits< DataType , Properties ... > {
   ~View() {}
```
Comment (mhoemmen) [inline core/src/Kokkos_View.hpp]:
```c++
KOKKOS_INLINE_FUNCTION
~View() = default;
```

File: core/src/Kokkos_View.hpp [hunk 2]
```diff
@@ -1985,7 +1985,7 @@ class View : public ViewTraits< DataType , Properties ... > {
...
 
   KOKKOS_INLINE_FUNCTION
-  View() : m_track(), m_map() { printf("empty constructor\n"); }
+  View() : m_track(), m_map() {  }
```
Comment (mhoemmen) [inline core/src/Kokkos_View.hpp]:
```c++
KOKKOS_INLINE_FUNCTION
View() = default;
```

File: core/src/Kokkos_View.hpp [hunk 3]
```diff
@@ -1995,26 +1995,24 @@ class View : public ViewTraits< DataType , Properties ... > {
...
 
        // Setup and initialization complete, start tracking
        m_track.assign_allocated_record_to_uninitialized( record );
-       printf("after duplicate record assigned to tracker: %d \n", m_track.tracking_disabled());
      }
 #endif
 
   }
 
   KOKKOS_INLINE_FUNCTION
```
Comment (mhoemmen) [inline core/src/Kokkos_View.hpp]: You should be able to `default` all of these copy and move constructors and copy and move assignment operators.

**Comment by jeffmiles63** [comment]
closing...will open another.

**Pull request closed without merging**